src/commands/index.ts

```diff
@@ -215,55 +234,86 @@ export const setToken = vscode.commands.registerCommand(
-      }
-      text = replaceYear(year, license.key, text);
-    }
-
-    if (author !== "") {
-      text = replaceAuthor(author, license.key, text);
-    }
-
     const extension: string =
       vscode.workspace.getConfiguration("license").get("extension") ?? "";
 
     const filename: string =
       vscode.workspace.getConfiguration("license").get("filename") ?? "LICENSE";
 
-    const content = new TextEncoder().encode(text);
-
     const licensePath = vscode.Uri.file(
       `${folder.uri.fsPath}/${filename}${
         multiple ? `-${license.spdx_id?.toUpperCase()}` : ""
       }${extension}`
     );
 
-    try {
-      await vscode.workspace.fs.stat(licensePath);
-
-      const answer = await vscode.window.showInformationMessage(
-        "License file already exists in this folder. Override it?",
-        "Yes",
-        "No"
-      );
-
-      if (answer === "Yes") {
-        await vscode.workspace.fs.writeFile(licensePath, content);
+    let year: string = vscode.workspace
+      .getConfiguration("license")
+      .get("year")!;
+
+    // DONE: feat: autoupdate license year if set to auto. fix issue #578
+    if (year === "auto") {
+      // context.workspaceState.update(license.key, 1997); // test --> 1997-2023
+      // about workspaceState, refer https://code.visualstudio.com/api/references/vscode-api#Memento
+      let prevYear = context.workspaceState.get<number>(license.key);
+      let currYear = new Date().getFullYear();
+      if (prevYear === undefined || prevYear === currYear) {
+        year = currYear + "";
+      } else {
+        year = `${prevYear}-${currYear}`;
       }
-    } catch {
-      await vscode.workspace.fs.writeFile(licensePath, content);
+    }
+    text = replaceYear(year, license.key, text);
+
+    // DONE: feat: autoassign author from git. fix issue #579
+    // get author by 'git config --global --get user.name'
+    let author: string | undefined = vscode.workspace
+      .getConfiguration("license")
+      .get("author");
+
+    // get author from .gitconfig
+    const gitExtension =
+      vscode.extensions.getExtension<GitExtension>("vscode.git")!.exports;
+    const git = gitExtension.getAPI(1);
+
+    // folder must be a git repo
+    const repo = git.getRepository(folder.uri);
+    if (!repo) {
+      vscode.window.showErrorMessage("Please use git init.");
+      return;
+    }
+
+    if (!author) {
+      // local
+      repo
+        .getConfig("user.name")
+        .then((localAuthor) => {
+          text = replaceAuthor(localAuthor, license.key, text);
+          generate(licensePath, text);
+        })
+        .catch(() => {
+          // global
+          repo
+            .getGlobalConfig("user.name")
+            .then((globalAuthor) => {
+              text = replaceAuthor(globalAuthor, license.key, text);
+              generate(licensePath, text);
+            })
+            .catch(() => {
+              vscode.commands.executeCommand("license.setAuthor");
+            });
```

Comment:
`await` and `try/catch` can be used here instead of callbacks.

Reply:
my bad.